apps/api-extractor/src/analyzer/ExportAnalyzer.ts

```diff
@@ -291,6 +297,33 @@ export class ExportAnalyzer {
     return true;
   }
 
+  /**
+   * Returns true if the module specifier matches a path mapping entry in the tsconfig `paths` map.
+   *
+   * @remarks
+   * Handles two types of path mappings:
+   *
+   * - Simple path-like strings: `some/path/to/import`
+   * - Paths with wildcards at the end: `some/path/to/*`
+   */
+  private _hasPathMappingMatch(moduleSpecifier: string): boolean {
+    const pathKeys: string[] = Object.keys(this._program.getCompilerOptions().paths || {});
```

Comment:
`_hasPathMappingMatch()` will get called for every single import, possibly multiple times.  If a project explicitly mapped each path for some reason, the array might contain hundreds of items.

A possible optimization:
- Keep the non-wildcard paths in a `Set<string>` rather than constructing+iterating an array for each call
- Keep the wildcard prefixes in an array, but reuse the same array

Reply:
Followed your advice... mostly. I'm still constructing an array for every call in order to determine whether any path mappings even exist. This could be expensive for projects with lots of path mappings. However, without this logic, then we'll attempt to initialize the `_exactPathMappings` and `_prefixPathMappings` data structures on every call and can't return out early for projects with no path mappings. Given most projects will have no path mappings, I opted for the current approach, but I don't feel strongly either way.

Reply:
This thread is no longer applicable given the new approach.